src/components/Identity/createComponent.js

```diff
@@ -82,7 +94,18 @@ export default ({
           return consent
             .withConsent()
             .then(() => {
-              return namespaces ? undefined : getIdentity(options);
+              if (namespaces) {
+                return undefined;
```

Comment:
?This section seems identical with the section from line 67. Can this be extracted as a method